app/experimenter/nimbus-ui/src/components/PageEditOverview/index.tsx

```diff
@@ -2,43 +2,85 @@
 import PageEditContainer from "../PageEditContainer";
+import { useMutation } from "@apollo/client";
+import { UPDATE_EXPERIMENT_OVERVIEW_MUTATION } from "../../gql/experiments";
+import { SUBMIT_ERROR } from "../../lib/constants";
+import { UpdateExperimentInput } from "../../types/globalTypes";
+import { updateExperimentOverview_updateExperimentOverview as UpdateExperimentOverviewResult } from "../../types/updateExperimentOverview";
+import { getExperiment_experimentBySlug } from "../../types/getExperiment";
 
 type PageEditOverviewProps = {} & RouteComponentProps;
 
 const PageEditOverview: React.FunctionComponent<PageEditOverviewProps> = () => {
-  // TODO: EXP-462 Get this from constants / config loaded at app start?
-  const applications = ["firefox-desktop", "fenix", "reference-browser"];
+  const [updateExperimentOverview, { loading }] = useMutation<
+    { updateExperimentOverview: UpdateExperimentOverviewResult },
+    { input: UpdateExperimentInput }
+  >(UPDATE_EXPERIMENT_OVERVIEW_MUTATION);
 
-  const [submitErrors /* setSubmitErrors */] = useState<Record<string, any>>(
-    {},
-  );
+  const [submitErrors, setSubmitErrors] = useState<Record<string, any>>({});
+  const currentExperiment = useRef<getExperiment_experimentBySlug>();
 
-  const onFormSubmit = useCallback(() => {
-    console.log("SUBMIT TBD");
-  }, []);
+  const onFormSubmit = useCallback(
+    async (
+      { name, hypothesis, application, publicDescription }: Record<string, any>,
+      resetForm: Function,
+    ) => {
+      try {
+        const result = await updateExperimentOverview({
+          variables: {
+            input: {
+              id: currentExperiment.current!.id,
+              name,
+              hypothesis,
+              application,
+              publicDescription,
+            },
+          },
+        });
+
+        if (!result.data?.updateExperimentOverview) {
+          throw new Error("Save failed, no error available");
+        }
+
+        const { message } = result.data.updateExperimentOverview;
+
+        if (message !== "success" && typeof message === "object") {
+          return void setSubmitErrors(message);
+        }
+
+        resetForm({ name, hypothesis, application, publicDescription });
+      } catch (error) {
+        setSubmitErrors({ "*": SUBMIT_ERROR });
+      }
+    },
+    [updateExperimentOverview, currentExperiment],
+  );
 
   const onFormNext = useCallback(() => {
-    console.log("NEXT TBD");
+    navigate(`branches`);
   }, []);
 
   return (
     <PageEditContainer title="Overview" testId="PageEditOverview">
-      {({ experiment }) => (
-        <FormOverview
-          {...{
-            isLoading: false,
-            applications,
-            experiment,
-            submitErrors,
-            onSubmit: onFormSubmit,
-            onNext: onFormNext,
-          }}
-        />
-      )}
+      {({ experiment }) => {
+        currentExperiment.current = experiment;
```

Comment:
Hmm, feeling a little leery of this bit of ref-based state management. But, I can't think of any clear problems this would cause.

Reply:
I only did this so I could define `onFormSubmit` outside of the render; but happy remove this and move it back in?